src/App.jsx

```diff
@@ -7,15 +7,22 @@ function App() {
   const [todoList, setTodoList] = useState([]);
 
   function addTodo(title) {
-    const newTodo = { id: Date.now(), title };
+    const newTodo = { id: Date.now(), title, isCompleted: false };
     setTodoList([...todoList, newTodo]);
   }
 
+  function completeTodo(id) {
+    const updatedTodos = todoList.map((todo) =>
+      todo.id === id ? { ...todo, isCompleted: true } : todo
+    );
+    setTodoList(updatedTodos);
```

Comment:
This is a nice catch. `updatedTodos` is already a copy of `todoList` so no need to create another copy before passing it to the setter function.